src/recorder/jamrecorder.cpp

```diff
@@ -88,8 +89,12 @@ void CJamClient::Frame(const QString _name, const CVector<int16_t>& pcm, int iSe
  */
 void CJamClient::Disconnect()
 {
-    static_cast<CWaveStream*>(out)->finalise();
-    out = nullptr;
+    if (out)
+    {
+        static_cast<CWaveStream*>(out)->finalise();
```

Comment:
This waits, right?  So long as that's true, we're okay...

Reply:
Again, it's the same as the old code, except that:
1. It checks `out` before using it, and
2. It frees what `out` was pointing to before discarding the value of `out`. That was the other main leak.

Reply:
Ah sorry, yes.